segmatch/src/segmented_cloud.cpp

```diff
@@ -187,6 +187,18 @@ bool SegmentedCloud::findValidSegmentById(const Id segment_id, Segment* result)
   }
 }
 
+bool SegmentedCloud::findValidSegmentPtrById(const Id segment_id, Segment** result) {
+  if (!findValidSegmentById(segment_id, NULL)) { return false; }
+  if (result != NULL) { *result = CHECK_NOTNULL(&valid_segments_.at(segment_id)); }
+  return true;
+}
+
+Segment SegmentedCloud::getValidSegmentByIndex(const size_t index) const {
+  CHECK_LT(index, valid_segments_.size()) <<
+      "Attempted to access index out of bounds of valid_segments_";
+  return valid_segments_.at(index);
```

Comment:
`valid_segments_` is now an unordered map, this would give you the element with Id value of index. I think we can do without this function.